modules/engine/src/index.ts

```diff
@@ -899,6 +908,64 @@ export class VectorEngine implements IVectorEngine {
     }
     this.logger.info({ transfer, method, methodId }, "Transfer pre-resolve");
 
+    // special case for crosschain transfer
+    // we need to generate a separate sig for withdrawal commitment since the transfer resolver may have gotten forwarded
+    // and needs to be regenerated for this leg of the transfer
+    const isCrossChain = await isCrosschainTransfer(transfer, this.chainAddresses, this.chainService);
+    if (isCrossChain) {
+      // first check if the provided sig is valid. in the case of the receiver directly resolving the withdrawal, it will
+      // be valid already
+      let channel: FullChannelState | undefined;
+      try {
+        channel = await this.store.getChannelState(transfer.channelAddress);
+      } catch (e) {
+        return Result.fail(
+          new RpcError(RpcError.reasons.ChannelNotFound, transfer.channelAddress, this.publicIdentifier, {
+            getChannelStateError: jsonifyError(e),
+          }),
+        );
+      }
+      if (!channel) {
+        return Result.fail(
+          new RpcError(RpcError.reasons.ChannelNotFound, transfer.channelAddress, this.publicIdentifier),
+        );
+      }
+      const {
+        transferState: { nonce, initiatorSignature, fee, callTo, callData, balance },
+      } = transfer;
+      const withdrawalAmount = balance.amount.reduce((prev, curr) => prev.add(curr), BigNumber.from(0)).sub(fee);
+      const commitment = new WithdrawCommitment(
+        channel.channelAddress,
+        channel.alice,
+        channel.bob,
+        transfer.balance.amount[0],
+        transfer.assetId,
+        withdrawalAmount.toString(),
+        nonce,
+        callTo,
+        callData,
+      );
+      let recovered: string;
+      try {
+        recovered = await recoverAddressFromChannelMessage(
+          commitment.hashToSign(),
+          params.transferResolver.responderSignature,
+        );
+      } catch (e) {
+        recovered = e.message;
+      }
```

Comment:
This type of checking should be in the `paramConverter.ts`

Reply:
Done.